numpy/core/src/multiarray/einsum.c.src

```diff
@@ -2092,83 +2091,101 @@ get_single_op_view(PyArrayObject *op, int  iop, char *labels,
     return 0;
 }
 
+static int
+_no_labels_are_negative(signed char *labels, int ndim)
+{
+    int idim;
+
+    for (idim = 0; idim < ndim; ++idim) {
+        /*
+         * The char type may be either signed or unsigned, we
+         * need it to be signed here.
+         */
+        int label = labels[idim];
+
+        if (label < 0) {
```

Comment:
Could just write `if (labels[idim] < 0)` here, now that there's no casting to worry about